pdl-compiler/scripts/generate_python_backend.py

```diff
@@ -1012,17 +1012,34 @@ def generate_enum_declaration(decl: ast.EnumDeclaration) -> str:
     enum_name = decl.id
     tag_decls = []
     for t in decl.tags:
+        # Enums in python are closed and ranges cannot be represented;
+        # instead the generated code uses Union[int, Enum]
+        # when ranges are used.
         if t.value is not None:
             tag_decls.append(f"{t.id} = {hex(t.value)}")
+
+    tag_ranges = []
+    for t in decl.tags:
         if t.range is not None:
-            tag_decls.append(f"{t.id}_START = {hex(t.range[0])}")
-            tag_decls.append(f"{t.id}_END = {hex(t.range[1])}")
+            tag_ranges.append(f"if v >= 0x{t.range[0]:x} and v <= 0x{t.range[1]:x}:")
+            tag_ranges.append(f"    return v")
 
     return dedent("""\
 
         class {enum_name}(enum.IntEnum):
             {tag_decls}
-        """).format(enum_name=enum_name, tag_decls=indent(tag_decls, 1))
+
+            @staticmethod
+            def from_int(v: int) -> Union[int, '{enum_name}']:
```

Comment:
nit: from_int doesn't convey the meaning that this will raise an exception for invalid values, should we rename this to something like `parse` or `validate` ?

Reply:
I figured it would be implicit since IntEnum constructor also raises an exception for invalid values

Reply:
Ack